### PR TITLE
style: Isortifying the repo and adding pyproject.toml

### PR DESCRIPTION
**Pull request by vrdmr**
## Description

Adding Isort for the repo and adding configuration.

- Removed the isort.cfg in favor of PyProject.toml file
- Isorted the complete azure_functions_worker and test module. Made minor changes in the sort order and skipped protos.

Fixes #

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/linters/tox.ini [hunk 1]
```diff
@@ -32,7 +32,7 @@ exclude =
         venv*,
         scripts
 
-max-line-length = 80
+max-line-length = 88
```
Comment (vrdmr) [inline .github/linters/tox.ini:35]: tox.ini max length is used by super linter.

File: .isort.cfg
Comment (vrdmr) [inline .isort.cfg:1]: No need for isort specific config, using `pyproject.toml` to set these settings.

File: azure_functions_worker/bindings/__init__.py [hunk 2]
```diff
@@ -1,18 +1,22 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-from .tracecontext import TraceContext
-from .retrycontext import RetryContext
+from .retrycontext import RetryContext  # isort: skip
+from .tracecontext import TraceContext  # isort: skip
```
Comment (vrdmr) [inline azure_functions_worker/bindings/__init__.py:4]: Skipping these two for preventing cyclic import issue when importing `from .context import Context`

File: dev-setup.py
Comment (vrdmr) [inline dev-setup.py:1]: Eventually `setup.py` will be removed and moved to `tasks.py` for test setup. Commenting it for now.

File: pyproject.toml
Comment (vrdmr) [inline pyproject.toml:1]: New file - main change - new place to keep the version as well. (would need a change in our release pipelines)

File: dev-setup.py [hunk 3]
```diff
@@ -474,32 +474,32 @@ def run(self) -> None:
...
-    long_description=open("README.md").read(),
-    long_description_content_type="text/markdown",
-    classifiers=CLASSIFIERS,
-    license="MIT",
-    packages=PACKAGES,
-    install_requires=INSTALL_REQUIRES,
-    extras_require=EXTRA_REQUIRES,
-    include_package_data=True,
-    cmdclass=COMMAND_CLASS,
-    test_suite="tests",
```
Comment (hallvictoria) [inline dev-setup.py:504]: Should CLASSIFIERS, PACKAGES, INSTALL_REQUIRES, etc also be commented out? They're only used here. 

Also, should this be commented out or can it be removed?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+python37 = [
+    "protobuf~=3.19.3",
+    "grpcio-tools~=1.43.0",
+    "grpcio~=1.43.0"
+]
+python38_plus = [
+    "protobuf~=4.22.0",
+    "grpcio-tools~=1.54.2",
+    "grpcio~=1.54.2",
+    "azurefunctions-extensions-base"
```
Comment (hallvictoria) [inline pyproject.toml]: The python37 and python38_plus should be required dependencies. We can specify what to install based on the python version like so:

"protobuf~=3.19.3; python_version == '3.7'",
"protobuf~=4.22.0; python_version >= '3.8'"

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    "*.egg-info",
+    ".tox",
+    ".nox",
+    ".git",
+    "__pycache__",
+    ".ci",
+    ".devcontainer",
+    ".github",
+#    "eng",
+#    "pack",
```
Comment (hallvictoria) [inline pyproject.toml]: nit: remove the commented out ones?

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,115 @@
+[project]
+name = "azure-functions-worker"
+version = "4.29.0"
```
Comment (hallvictoria) [inline pyproject.toml]: We can make the version dynamic and pull it from azure_functions_worker/version.py

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    ".nox",
+    ".git",
+    "__pycache__",
+    ".ci",
+    ".devcontainer",
+    ".github",
+#    "eng",
+#    "pack",
+    "*protos/*"
+]
```
Comment (hallvictoria) [inline pyproject.toml]: nit: add [project.urls] for the github repo URL like we did in setup.py

File: pyproject.toml (hunk 6 above)
Comment (gavin-aguiar) [inline pyproject.toml]: Can we reference the version.py file in pyproject.toml? If not, lets remove version.py

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+package-dir = {"" = "azure_functions_worker"}
+
+[tool.isort]
+profile = "black"
```
Comment (gavin-aguiar) [inline pyproject.toml:92]: Is this needed?

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+line_length = 88
```
Comment (gavin-aguiar) [inline pyproject.toml:93]: This is different from the one in tox.ini right?

**Review by gavin-aguiar — APPROVED** [review]

**Review by hallvictoria — APPROVED** [review]

**Pull request merged**